sdk/tables/Azure.Data.Tables/src/TableBatchResponse.cs

```diff
@@ -32,12 +31,46 @@ internal TableBatchResponse(ConcurrentDictionary<string, (HttpMessage Message, R
         /// <returns></returns>
         public Response GetResponseForEntity(string rowKey)
         {
-            if (!_requestLookup.TryGetValue(rowKey, out (HttpMessage Message, RequestType RequestType) tuple))
+            if (!_requestLookup.TryGetValue(rowKey, out HttpMessage message))
             {
                 throw new InvalidOperationException("The batch operation did not contain an entity with the specified rowKey");
             }
 
-            return tuple.Message.Response;
+            return message.Response;
+        }
+
+        /// <summary>
+        /// Tries to get the entity that caused the batch operation failure from the <see cref="RequestFailedException"/>.
+        /// </summary>
+        /// <param name="exception">The exception thrown from <see cref="TableClient.SubmitTransaction"/> or <see cref="TableClient.SubmitTransactionAsync"/>.</param>
+        /// <param name="submittedBatchItems">The submitted list of <see cref="BatchItem"/>. This list should be unchanged since it was submitted via
+        /// <see cref="TableClient.SubmitTransaction"/> or <see cref="TableClient.SubmitTransactionAsync"/> </param>
+        /// <param name="failedEntity">If the return value is <c>true</c>, contains the <see cref="ITableEntity"/> that caused the batch operation to fail.</param>
+        /// <returns><c>true</c> if the failed entity was retrieved from the exception, else <c>false</c>.</returns>
+        public static bool TryGetFailedEntityFromException(RequestFailedException exception, List<BatchItem> submittedBatchItems, out ITableEntity failedEntity)
```

Comment:
Can we hold onto the `submittedBatchItems` internaly?

Reply:
This is going to change based on arch board feedback. We're going to encapsulate all this into a custom exception type.